components/appmgt/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APPMgtUiActivitiesBamDataPublisher.java

```diff
@@ -232,20 +220,18 @@ public void publishUserActivityEvents(String action, String item,
 				}
 			} else {
 				// Write directly to DB
-				AppMDAO.saveStoreHits(appId.trim(), userId.trim(), tenantId,
-						strDataContext);
-			}
+                AppMDAO.saveStoreHits(appId.trim(), userId.trim(), tenantId);
```

Comment:
I seem to have misinterpreted this as an instance of AppMDAO class, but it looks like "saveStoreHits" is a static method, in which case, this is alright. However, the point you just mentioned still stands. Might be a better option to rename the class as AppManagerDAO.

One another point is that, if using static methods, better to make make this a util. Current naming of the class suggests that it is a DAO but this doesn't seem to adhere the standards enforced by typical DAO patterns. My personal preference would be to get this implemented adhering proper DAO pattern conventions.